web/tests/functional/component/test_component.py

```diff
@@ -583,3 +583,37 @@ def test_component_anywhere_on_path(self):
         self.assertEqual(len(component_results), 1)
         self.assertTrue(
             component_results[0].checkedFile.endswith('path_end.h'))
+
+    def test_full_report_path_in_component(self):
+        """
+        Check "full report path in component" feature. With this flag one can
+        query all reports where the bug path not only ends in the component,
+        but every point of the bug path is inside the component.
+        """
+        components = [
+            {
+                'name': 'main',
+                'value': '+*/same_origin.cpp'
+            },
+            {
+                'name': 'calculate',
+                'value': '+*/calculate.h'
+            }
+        ]
+
+        for c in components:
+            self.__add_new_component(c)
+
+        self.assertEqual(len(components), 2)
+
+        r_filter = ReportFilter(componentNames=['calculate'])
+        component_results = self._cc_client.getRunResults(
+            None, 500, 0, None, r_filter, None, False)
+        self.assertEqual(len(component_results), 2)
```

Comment:
which are the ones we are expecting here?
Can you maybe add them in comments or string match the output of the query?
maybe also write out explicitly: fullReportPathInComponent=False

'clang-diagnostic-division-by-zero'
'core.divideZero'

?